Tests for copymode throwing NotImplementedError failing on macOS

```
_______________ test_copymode_local_to_local_symlink_dont_follow _______________

tmpdir = local('/private/tmp/nix-build-python3.9-smbprotocol-1.5.1.drv-0/pytest-of-veprbl/pytest-0/test_copymode_local_to_local_s1')

    @pytest.mark.skipif((os.name == 'nt' and sys.version_info[0] < 3) or
                        (sys.version_info[0] == 3 and sys.version_info[1] == 5),
                        reason="Python 3.5 errors out on os.chmod(follow_symlinks=False)")
    def test_copymode_local_to_local_symlink_dont_follow(tmpdir):
        test_dir = tmpdir.mkdir('test')
        src_filename = "%s\\source.txt" % test_dir
        dst_filename = "%s\\target.txt" % test_dir
    
        with open(src_filename, mode='w') as fd:
            fd.write(u"content")
        os.chmod(src_filename, stat.S_IREAD)
    
        with open(dst_filename, mode='w') as fd:
            fd.write(u"content")
    
        src_link = "%s\\source-link.txt" % test_dir
        dst_link = "%s\\target-link.txt" % test_dir
    
        os.symlink(src_filename, src_link)
        os.symlink(dst_filename, dst_link)
    
        expected = "chmod: follow_symlinks unavailable on this platform"
        with pytest.raises(NotImplementedError, match=re.escape(expected)):
>           copymode(src_link, dst_link, follow_symlinks=False)
E           Failed: DID NOT RAISE <class 'NotImplementedError'>

tests/test_smbclient_shutil.py:605: Failed
____________ test_copystat_local_to_local_symlink_dont_follow_fail _____________

tmpdir = local('/private/tmp/nix-build-python3.9-smbprotocol-1.5.1.drv-0/pytest-of-veprbl/pytest-0/test_copystat_local_to_local_s1')

    @pytest.mark.skipif((os.name == 'nt' and sys.version_info[0] < 3) or
                        (sys.version_info[0] == 3 and sys.version_info[1] == 5),
                        reason="Python 3.5 errors out on os.chmod(follow_symlinks=False)")
    def test_copystat_local_to_local_symlink_dont_follow_fail(tmpdir):
        test_dir = tmpdir.mkdir('test')
        src_filename = "%s\\source.txt" % test_dir
        dst_filename = "%s\\target.txt" % test_dir
    
        with open(src_filename, mode='w') as fd:
            fd.write(u"content")
        os.chmod(src_filename, stat.S_IREAD)
    
        with open(dst_filename, mode='w') as fd:
            fd.write(u"content")
    
        src_link = "%s\\source-link.txt" % test_dir
        dst_link = "%s\\target-link.txt" % test_dir
    
        os.symlink(src_filename, src_link)
        os.symlink(dst_filename, dst_link)
    
        expected = "follow_symlinks unavailable on this platform"
        with pytest.raises(NotImplementedError, match=re.escape(expected)):
>           copystat(src_link, dst_link, follow_symlinks=False)
E           Failed: DID NOT RAISE <class 'NotImplementedError'>
```

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -1,3 +1,7 @@
+#
+# Available MS hosted OSes
+# https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted
+#
 trigger:
   branches:
     include:
@@ -25,7 +29,7 @@ stages:
 
   - job: Test_Linux
     pool:
-      vmImage: ubuntu-18.04
+      vmImage: $(vmImage)
     variables:
       smb_port: 445
       smb_user: smbuser
@@ -35,12 +39,19 @@ stages:
       matrix:
         Python36:
           python.version: 3.6
+          vmImage: ubuntu-16.04
         Python37:
           python.version: 3.7
+          vmImage: ubuntu-20.04
         Python38:
           python.version: 3.8
+          vmImage: ubuntu-20.04
         Python39:
           python.version: 3.9
+          vmImage: ubuntu-20.04
+        Python39MacOs:
+          python.version: 3.9
+          vmImage: macOS-10.15
 
     steps:
     - task: UsePythonVersion@0

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -1,4 +1,7 @@
 #
+# Docs for YAML schema
+# https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema
+#
 # Available MS hosted OSes
 # https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted
 #
@@ -29,7 +32,7 @@ stages:
 
   - job: Test_Linux
     pool:
-      vmImage: $(vmImage)
+      vmImage: ubuntu-18.04
     variables:
       smb_port: 445
       smb_user: smbuser
@@ -39,19 +42,12 @@ stages:
       matrix:
         Python36:
           python.version: 3.6
-          vmImage: ubuntu-16.04
         Python37:
           python.version: 3.7
-          vmImage: ubuntu-20.04
         Python38:
           python.version: 3.8
-          vmImage: ubuntu-20.04
         Python39:
           python.version: 3.9
-          vmImage: ubuntu-20.04
-        Python39MacOs:
-          python.version: 3.9
-          vmImage: macOS-10.15
 
     steps:
     - task: UsePythonVersion@0
@@ -98,6 +94,49 @@ stages:
         SMB_USER: $(smb_user)
         SMB_PASSWORD: $(smb_pass)
         SMB_SHARE: $(smb_share)
+      displayName: Test
+
+    - task: PublishTestResults@2
+      inputs:
+        testResultsFiles: '**/test-*.xml'
+        testRunTitle: $(Agent.OS) - $(Build.BuildNumber) - Python $(python.version)
+      displayName: Publish test results
+      condition: succeededOrFailed()
+
+    - task: PublishCodeCoverageResults@1
+      inputs:
+        codeCoverageTool: Cobertura
+        summaryFileLocation: $(System.DefaultWorkingDirectory)/**/coverage.xml
+
+    - script: |
+        bash <(curl -s https://codecov.io/bash)
+      displayName: Upload to codecov.io
+      continueOnError: true
+      timeoutInMinutes: 5
+
+
+  - job: Test_MacOs
+    pool:
+      vmImage: macOS-10.15
+
+    steps:
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: 3.9
+
+    - script: |
+        echo "Installing baseline pip packages"
+        python -m pip install --upgrade pip setuptools wheel coverage -c tests/constraints.txt
+
+        echo "Installing test requirements"
+        python -m pip install -r requirements-test.txt -c tests/constraints.txt
+
+        echo "Installing package"
+        python -m pip install . -c tests/constraints.txt
+      displayName: Install
+
+    - script: |
+        pytest -v --junitxml junit/test-results.xml --cov smbclient --cov smbprotocol --cov-report xml --cov-report term-missing
       displayName: Test
 
     - task: PublishTestResults@2

--- a/tests/test_smbclient_shutil.py
+++ b/tests/test_smbclient_shutil.py
@@ -605,6 +605,8 @@ def test_copymode_local_to_local_symlink_follow(tmpdir):
     assert stat.S_IMODE(actual_link) & stat.S_IWRITE == stat.S_IWRITE
 
 
+@pytest.mark.skipif(sys.platform.startswith('darwin'),
+                    reason="On macOS os.chmod supports symlinks.")
 def test_copymode_local_to_local_symlink_dont_follow(tmpdir):
     test_dir = tmpdir.mkdir('test')
     src_filename = "%s\\source.txt" % test_dir
@@ -858,6 +860,8 @@ def test_copystat_local_to_local_symlink_follow(tmpdir):
     assert stat.S_IMODE(actual_link.st_mode) & stat.S_IWRITE == stat.S_IWRITE
 
 
+@pytest.mark.skipif(sys.platform.startswith('darwin'),
+                    reason="On macOS os.chmod supports symlinks.")
 def test_copystat_local_to_local_symlink_dont_follow_fail(tmpdir):
     test_dir = tmpdir.mkdir('test')
     src_filename = "%s\\source.txt" % test_dir

